source/proximity_list/proximity_list.cpp

```diff
@@ -1,9 +1,69 @@
 #include "proximity_list.h"
+#include "type/type.h"
+#include "utils/logger.h"
+#include "flann/flann.hpp"
+
+#include <cmath>
+#include <array>
+#include <exception>
 
 namespace navitia {
 namespace proximitylist {
+using type::GeographicalCoord;
+using index_t = flann::Index<flann::L2<float>>;
+
+template <class T>
+std::vector<std::pair<T, GeographicalCoord>> ProximityList<T>::find_within(GeographicalCoord coord,
+                                                                           double radius) const {
+    std::vector<std::pair<T, GeographicalCoord>> result;
+
+    log4cplus::Logger logger = log4cplus::Logger::getInstance("log");
+
+    if (!index) {
+        flann::Matrix<float> points{&data[0], data.size() / 3, 3};
+        index = std::make_shared<index_t>(points, flann::KDTreeSingleIndexParams(10));
+
+        LOG4CPLUS_INFO(logger, " Building NN Index!!!!!!!!!!");
+
+        index->buildIndex();
```

Comment:
this should be done before, at binarisation or after deserialization

Reply:
i know i know, i was lazy....